doc/io_streams.rdoc

```diff
@@ -0,0 +1,356 @@
+
+  # Text with 1-byte characters.
+  File.new('t.txt') {|f| f.gets(1) }  # => "F"
+  File.new('t.txt') {|f| f.gets(2) }  # => "Fi"
+  File.new('t.txt') {|f| f.gets(3) }  # => "Fir"
+  File.new('t.txt') {|f| f.gets(4) }  # => "Firs"
+  # No more than one line.
+  File.new('t.txt') {|f| f.gets(10) } # => "First line"
+  File.new('t.txt') {|f| f.gets(11) } # => "First line\n"
+  File.new('t.txt') {|f| f.gets(12) } # => "First line\n"
+
+  # Text with 2-byte characters, which will not be split.
+  File.new('r.rus') {|f| f.gets(1).size } # => 1
+  File.new('r.rus') {|f| f.gets(2).size } # => 1
+  File.new('r.rus') {|f| f.gets(3).size } # => 2
+  File.new('r.rus') {|f| f.gets(4).size } # => 2
+
+==== Line Separator and Line Limit
+
+With arguments +sep+ and +limit+ given,
+combines the two behaviors:
+
+- Returns the next line as determined by line separator +sep+.
+- But returns no more bytes than are allowed by the limit.
+
+Example:
+
+  File.new('t.txt') {|f| f.gets('li', 20) } # => "First li"
+  File.new('t.txt') {|f| f.gets('li', 2) }  # => "Fi"
+
+==== Line Number
+
+A readable \IO stream has a _line_ _number_,
+which is the non-negative integer line number
+in the stream where the next read will occur.
+
+A new stream is initially has line number +0+.
+
+\Method IO#lineno returns the line number.
+
+Reading lines from a stream usually changes its line number:
+
+  f = File.new('t.txt', 'r')
+  f.lineno   # => 0
+  f.readline # => "This is line one.\n"
+  f.lineno   # => 1
+  f.readline # => "This is the second line.\n"
+  f.lineno   # => 2
+  f.readline # => "Here's the third line.\n"
+  f.lineno   # => 3
+  f.eof?     # => true
+  f.close
+
+Iterating over lines in a stream usually changes its line number:
+
+   f = File.new('t.txt')
+   f.each_line do |line|
+     p "position=#{f.pos} eof?=#{f.eof?} lineno=#{f.lineno}"
+   end
+   f.close
+
+Output:
+
+  "position=11 eof?=false lineno=1"
+  "position=23 eof?=false lineno=2"
+  "position=24 eof?=false lineno=3"
+  "position=36 eof?=false lineno=4"
+  "position=47 eof?=true lineno=5"
+
+==== Line Options
+
+A number of \IO methods accept optional keyword arguments
+that determine how lines in a stream are to be treated:
+
+- +:chomp+: If +true+, line separators are omitted; default is +false+.
+
+=== Open and Closed \IO Streams
+
+A new \IO stream may be open for reading, open for writing, or both.
+
+You can close a stream using these methods:
+
+- +#close+ - closes the stream for both reading and writing.
+
+- +#close_read+ (not available in \ARGF) - closes the stream for reading.
+
+- +#close_write+ (not available in \ARGF) - closes the stream for writing.
+
+You can query whether a stream is closed using these methods:
+
+- +#closed?+ - returns whether the stream is closed.
+
+- +#closed_read?+ (\StringIO only) - returns whether the stream
+  is closed for reading.
+
+- +#closed_write?+ (\StringIO only) - returns whether the stream
+  is closed for writing.
```

Comment:
For features only available for a specific class (StringIO in this case), IMO we should not include that in this doc since this doc is for shared features.

Reply:
Removed.